File: mRemoteNG/Forms/OptionsPages/StartupExitPage.vb
Imports mRemoteNG.My

Namespace Forms.OptionsPages
    Public Class StartupExitPage
        Public Overrides Property PageName() As String
            Get
                Return Language.strStartupExit
            End Get
            Set(value As String)
            End Set
        End Property

        Public Overrides Sub ApplyLanguage()
            MyBase.ApplyLanguage()

            chkSaveConsOnExit.Text = Language.strSaveConsOnExit
            chkReconnectOnStart.Text = Language.strReconnectAtStartup
            chkSingleInstance.Text = Language.strAllowOnlySingleInstance
            chkProperInstallationOfComponentsAtStartup.Text = Language.strCheckProperInstallationOfComponentsAtStartup
        End Sub

        Public Overrides Sub SaveSettings()
            MyBase.SaveSettings()

            Settings.SaveConsOnExit = chkSaveConsOnExit.Checked
            Settings.OpenConsFromLastSession = chkReconnectOnStart.Checked
            Settings.SingleInstance = chkSingleInstance.Checked
            Settings.StartupComponentsCheck = chkProperInstallationOfComponentsAtStartup.Checked
        End Sub

        Private Sub StartupExitPage_Load(sender As System.Object, e As EventArgs) Handles MyBase.Load
            chkSaveConsOnExit.Checked = Settings.SaveConsOnExit
            chkReconnectOnStart.Checked = Settings.OpenConsFromLastSession
            chkSingleInstance.Checked = Settings.SingleInstance
            chkProperInstallationOfComponentsAtStartup.Checked = Settings.StartupComponentsCheck
        End Sub
    End Class
End Namespace

File: mRemoteV1.ConvertedToC#/Forms/OptionsPages/StartupExitPage.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using mRemoteNG.My;

namespace mRemoteNG.Forms.OptionsPages
{
	public partial class StartupExitPage
	{
		public override string PageName {
			get { return Language.strStartupExit; }
			set { }
		}

		public override void ApplyLanguage()
		{
			base.ApplyLanguage();

			chkSaveConsOnExit.Text = Language.strSaveConsOnExit;
			chkReconnectOnStart.Text = Language.strReconnectAtStartup;
			chkSingleInstance.Text = Language.strAllowOnlySingleInstance;
			chkProperInstallationOfComponentsAtStartup.Text = Language.strCheckProperInstallationOfComponentsAtStartup;
		}

		public override void SaveSettings()
		{
			base.SaveSettings();

			Settings.SaveConsOnExit = chkSaveConsOnExit.Checked;
			Settings.OpenConsFromLastSession = chkReconnectOnStart.Checked;
			Settings.SingleInstance = chkSingleInstance.Checked;
			Settings.StartupComponentsCheck = chkProperInstallationOfComponentsAtStartup.Checked;
		}

		private void StartupExitPage_Load(System.Object sender, EventArgs e)
		{
			chkSaveConsOnExit.Checked = Settings.SaveConsOnExit;
			chkReconnectOnStart.Checked = Settings.OpenConsFromLastSession;
			chkSingleInstance.Checked = Settings.SingleInstance;
			chkProperInstallationOfComponentsAtStartup.Checked = Settings.StartupComponentsCheck;
		}
		public StartupExitPage()
		{
			Load += StartupExitPage_Load;
			InitializeComponent();
		}
	}
}
